chore: add docs page for custom envs (#2386)
aws/copilot-cli: site/content/docs/developing/custom-environment-resources.en.md

# Custom Environment Resources

When creating a new [environment](../../concepts/environments.en.md) with Copilot, you are given the option to import existing VPC resources. (Use [flags with `env init`](../commands/env-init.en.md) or the guided experience, shown below.)
```bash
% copilot env init
What is your environment's name? env-name
Which credentials would you like to use to create name? [profile default]

  Would you like to use the default configuration for a new environment?
    - A new VPC with 2 AZs, 2 public subnets and 2 private subnets
    - A new ECS Cluster
    - New IAM Roles to manage services and jobs in your environment
  [Use arrows to move, type to filter]
    Yes, use default.
    Yes, but I'd like configure the default resources (CIDR ranges).
  > No, I'd like to import existing resources (VPC, subnets).
```

> When you select the default configuration, Copilot follows [AWS best practices](https://aws.amazon.com/blogs/containers/amazon-ecs-availability-best-practices/) and creates a VPC with two Availability Zones, each with one public and one private subnet. While this is what AWS recommends, Copilot allows some flexibility when you import your own resources: you may bring a VPC with only two private subnets and no public subnets for your workloads that are not internet-facing. (For more details on the resources you'll need for isolated networks, go [here](https://github.com/aws/copilot-cli/discussions/2378).)

Reviewer: ```suggestion
When you select the default configuration, Copilot follows [AWS best practices](https://aws.amazon.com/blogs/containers/amazon-ecs-availability-best-practices/) and creates a VPC with two Availability Zones, each with two public and two private subnets. While this is a good configuration for most cases, Copilot allows some flexibility when you import your own resources. For example, you may bring a VPC with only two private subnets and no public subnets for your workloads which are not internet-facing. (For more details on the resources you'll need for isolated networks, go [here](https://github.com/aws/copilot-cli/discussions/2378).)
```

Author: Thanks! But isn't it two public and two private total, with one of each in each AZ? 
Maybe this is clearer: "...creates a VPC with two public and two private subnets, with one of each type in an Availability Zone."??

Author: Pretty sure this has to be "that" not "which" because "are not internet-facing" is an essential clause. 🙂 🤓 
https://writingstudio.gsu.edu/files/2015/09/That-Which-Who-Clauses.pdf

Reviewer: > Maybe this is clearer: "...creates a VPC with two public and two private subnets, with one of each type in an Availability Zone."??

omg you're right - i misread and also don't understand the english language 

i'm so dumb

## Considerations
* If you are using a private hosted zone, [you must](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zone-private-considerations.html#hosted-zone-private-considerations-vpc-settings) set `enableDnsHostname` and `enableDnsSupport` to true.

> * To deploy internet-facing workloads in [private subnets](../include/common-svc-fields.md), your VPC will need a [NAT gateway](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-gateway.html). 

Author: Q: What happens if I link to the partial here? Better to pick a workload type and link to that page?

> * To deploy internet-facing workloads in [private subnets](../include/common-svc-fields.md), your VPC will need a [NAT gateway](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-gateway.html). 

Reviewer 2: Just thinking out loud: maybe in the future we can do a show-and-tell or a template example for creating a NAT gateway, and then put a link here.

Reviewer 2: ooo nvm! Users only need to create a NAT gateway if they are using their own private subnets right? If they are using Copilot-provisioned env resources they wouldn't have to worry about that. 

Is this `## Considerations` section more like a "Trouble Shooting" thing? Like in case users are bringing their own VPC resources and and then encounter issues that would have been dealt with by Copilot if they were using Copilot-provisioned ones.

Author: Yes, exactly. Just for imported VPC resources. I like the idea of doing a `private` placement show-and-tell as a partner to the `isolated` one!
